apps/server/src/shared/domain/rules/task.rule.ts

```diff
@@ -10,15 +10,17 @@ export class TaskRule extends BaseRule {
 		super();
 	}
 
-	hasPermission(user: User, task: Task, action: Actions): boolean {
-		const isCreator = this.hasAccessToEntity(user, task, ['creator']);
+	hasPermission(user: User, entity: Task, context: IPermissionContext): boolean {
+		const { action, requiredPermissions } = context;
+		const hasPermission = this.hasAllPermissions(user, requiredPermissions);
+		const isCreator = this.hasAccessToEntity(user, entity, ['creator']);
 		let hasCoursePermission = false;
-
-		if (task.course) {
-			hasCoursePermission = this.courseRule.hasPermission(user, task.course, action);
+		if (entity.course) {
+			const { course } = entity;
+			hasCoursePermission = this.courseRule.hasPermission(user, course, { action, requiredPermissions: [] });
 		}
-		const hasPermission = isCreator || hasCoursePermission;
+		const result = hasPermission && (isCreator || hasCoursePermission);
```

Comment:
We should think about hasPermission and creator what is needed for result true.

Reply:
If requiredPermissions equal [ ] -> hasPermission equal true